rego/rego.go

```diff
@@ -1645,9 +1645,10 @@ func (r *Rego) partialResult(ctx context.Context, pCfg *PrepareConfig) (PartialR
 	}
 
 	// Update compiler with partial evaluation output.
-	r.compiler.Modules["__partialresult__"] = module
+	id := fmt.Sprintf("__partialresult__%s__", ectx.partialNamespace)
+	r.compiler.Modules[id] = module
 	for i, module := range pq.Support {
-		r.compiler.Modules[fmt.Sprintf("__partialsupport%d__", i)] = module
+		r.compiler.Modules[fmt.Sprintf("__partialsupport__%s__%d__", ectx.partialNamespace, i)] = module
```

Comment:
I'm now wondering whether we should be doing this at all...this clearly fixes the issue but I'm questioning whether the rego package should modify the compiler it's been given. Perhaps we should start considering a `regov2` package. Just a thought.

Reply:
+100

I think its a nice user experience when you are not managing the compiler external to the `Rego` objects. Would definitely want to rethink some of this longer term.